src/datapilot/core/platforms/dbt/hooks/executor_hook.py

```diff
@@ -0,0 +1,59 @@
+import argparse
+import time
+from pathlib import Path
+from typing import Optional
+from typing import Sequence
+
+from datapilot.config.config import load_config
+from datapilot.core.platforms.dbt.constants import MODEL
+from datapilot.core.platforms.dbt.constants import PROJECT
+from datapilot.core.platforms.dbt.executor import DBTInsightGenerator
+from datapilot.utils.utils import generate_partial_manifest_catalog
+from datapilot.utils.utils import get_tmp_dir_path
+
+
+def main(argv: Optional[Sequence[str]] = None):
+    start_time = time.time()
+    parser = argparse.ArgumentParser()
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--config-path",
+        nargs="*",
+        help="Path of the config file to be used for the insight generation",
+    )
+
+    args = parser.parse_known_args(argv)
+
+    config = {}
+    if hasattr(args[0], "config_path") and args[0].config_path:
+        print(f"Using config file: {args[0].config_path[0]}")
+        config = load_config(args[0].config_path[0])
+
+    changed_files = args[1]
+    tmp_folder = get_tmp_dir_path()
```

Comment:
We should Ideally have args which user can provide for this. A way in which they can override

Reply:
Since we are going to be using the dbt_project.yml. We should pass the project root for it to work neatly